discovery_engine_core/ai/ai/src/utils.rs

```diff
@@ -73,6 +75,20 @@ pub(crate) fn system_time_now() -> SystemTime {
     SystemTime::now()
 }
 
+/// Rank documents based on their score.
+///
+/// # Panics
+/// This function will panic if a score for a document is missing.
+pub fn rank<D, S>(documents: &mut [D], scores: &HashMap<D::Id, f32, S>)
+where
+    D: Document,
+    S: std::hash::BuildHasher,
+{
+    documents.sort_unstable_by(|a, b| {
+        nan_safe_f32_cmp_desc(scores.get(a.id()).unwrap(), scores.get(b.id()).unwrap())
```

Comment:
🔧 if we unwrap anyways, we could just use index notation
```suggestion
        nan_safe_f32_cmp_desc(&scores[a.id()], &scores[b.id()])
```

Reply:
It will only make more obscure that we can panic here and more difficult to look for functions that can panic. Clippy don't even worn that the panic docs is missing.